praw/models/reddit/live.py

```diff
@@ -217,6 +128,141 @@ def remove_invite(self, redditor):
+            return other == str(self)
+        return (isinstance(other, self.__class__) and
+                str(self) == str(other))
+
+    def __getitem__(self, update_id):
+        """Return a lazy :class:`.LiveUpdate` instance.
+
+        .. warning:: At this time, accesing lazy attributes, whose value
+           have not loaded, raises ``AttributeError``.
+
+        :param update_id: A live update ID, e.g.,
+            ``'7827987a-c998-11e4-a0b9-22000b6a88d2'``.
+
+        Usage:
+
+        .. code-block:: python
+
+           thread = reddit.live('ukaeu1ik4sw5')
+           update = thread['7827987a-c998-11e4-a0b9-22000b6a88d2']
+           update.thread     # LiveThread(id='ukaeu1ik4sw5')
+           update.id         # '7827987a-c998-11e4-a0b9-22000b6a88d2'
+           update.author     # raise ``AttributeError``
+        """
+        return LiveUpdate(self._reddit, self.id, update_id)
+
+    def __hash__(self):
+        """Return the hash of the current instance."""
+        return hash(self.__class__.__name__) ^ hash(str(self))
+
+    def __init__(self, reddit, id=None,  # pylint: disable=redefined-builtin
+                 _data=None):
+        """Initialize a lazy :class:`.LiveThread` instance.
+
+        :param reddit: An instance of :class:`.Reddit`.
+        :param id: A live thread ID, e.g., ``'ukaeu1ik4sw5'``
+        """
+        if bool(id) == bool(_data):
+            raise TypeError('Either `id` or `_data` must be provided.')
+        super(LiveThread, self).__init__(reddit, _data)
+        if id:
+            self.id = id  # pylint: disable=invalid-name
+        self._contrib = None
+        self._contributor = None
+
+    def _info_path(self):
+        return API_PATH['liveabout'].format(id=self.id)
+
+    def updates(self, **generator_kwargs):
+        """Return a :class:`.ListingGenerator` yields :class:`.LiveUpdate` s.
+
+        :param generator_kwargs: keyword arguments passed to
+            :class:`.ListingGenerator` constructor.
+        :returns: A :class:`.ListingGenerator` object which yields
+            :class:`.LiveUpdate` object.
+        """
+        url = API_PATH['live_updates'].format(id=self.id)
+        for update in ListingGenerator(self._reddit, url,
+                                       **generator_kwargs):
+            update._thread = self
+            yield update
+
+
+class LiveThreadContribution(object):
+    """Provides a set of contribution functions to a LiveThread."""
+
+    def __init__(self, thread):
+        """Create an instance of :class:`.LiveThreadContribution`.
+
+        :param thread: An instance of :class:`.LiveThread`.
+
+        This instance can be retrieved through ``thread.contrib``
+        where thread is a :class:`.LiveThread` instance. E.g.,
+
+        .. code-block:: python
+
+           thread = reddit.live('ukaeu1ik4sw5')
+           thread.contrib.add_update('### update')
+
+        """
+        self.thread = thread
+
+    def add_update(self, body):
```

Comment:
Oooh. I get it now. These are actions that contributors can provide. Somehow I missed that connection. That's great to group these functions into contributor-specific actions.

Separately I now understand the reason for 'contrib' over 'contributor' because 'moderator' was shortened to 'mod'. The difference there, I think, is that many people refer to a moderator as a 'mod', however I really don't think similar exists for a contributor. With that said, I'm not opposed to `contrib` so long as it's documented, which it is.